compiler/il/OMRBlock.cpp

```diff
@@ -805,6 +807,687 @@ OMR::Block::splitWithGivenMethodSymbol(TR::ResolvedMethodSymbol *methodSymbol, T
+ */
+static void replaceNodesInSubtree(TR::Node *node, NodeTable *nodeInfo, TR::NodeChecklist &checklist)
+   {
+   if (checklist.contains(node))
+      return;
+
+   checklist.add(node);
+
+   for (int i = 0; i < node->getNumChildren(); ++i)
+      {
+      TR::Node *child = node->getChild(i);
+      auto entry = nodeInfo->find(child);
+      if (entry != nodeInfo->end())
+         {
+         node->setAndIncChild(i, entry->second.second);
+         child->decReferenceCount();
+         }
+      else
+         {
+         replaceNodesInSubtree(child, nodeInfo, checklist);
+         }
+      }
+   }
+
+/*
+ * Search for the nodes to be uncommoned from the split point till end of extended basic block
+ *
+ * @param Compilation object
+ * @param start tree top from where it needs to start searching
+ * @param table containing information about original nodes to be replaced with corresponding replacement node 
+ */
+static void replaceNodesInTrees(TR::Compilation * comp, TR::TreeTop *start, NodeTable *nodeInfo)
+   {
+   TR::NodeChecklist checklist(comp);
+   while (start)
+      {
+      if (start->getNode()->getOpCodeValue() == TR::BBStart && !start->getNode()->getBlock()->isExtensionOfPreviousBlock())
+         break;
+      replaceNodesInSubtree(start->getNode(), nodeInfo, checklist);
+      start = start->getNextTreeTop();
+      }
+   }
+
+/*
+ * Analyzes the node and list of unavailable registers to find a register pair which can be used to store the node
+ *
+ * @param Compilation object
+ * @param Node which requires to be uncommoned
+ * @param BitVector constaining information about the unavailable registers
+ * @return Pair of GlobalRegisterNumber which can be used to store node
+ */
+static std::pair<TR_GlobalRegisterNumber,TR_GlobalRegisterNumber> findAvailableRegister(TR::Compilation *comp, TR::Node *node, TR_BitVector &unavailableRegisters)
+   {
+   TR_GlobalRegisterNumber firstGPR = comp->cg()->getFirstGlobalGPR();
+   TR_GlobalRegisterNumber lastGPR = comp->cg()->getLastGlobalGPR();
+   TR_GlobalRegisterNumber firstFPR = comp->cg()->getFirstGlobalFPR();
+   TR_GlobalRegisterNumber lastFPR = comp->cg()->getLastGlobalFPR();
+   TR_GlobalRegisterNumber firstVRF = comp->cg()->getFirstGlobalVRF();
+   TR_GlobalRegisterNumber lastVRF = comp->cg()->getLastGlobalVRF();
+
+   std::pair<TR_GlobalRegisterNumber,TR_GlobalRegisterNumber> toReturn = std::make_pair<TR_GlobalRegisterNumber,TR_GlobalRegisterNumber>(-1,-1);
+
+   TR_GlobalRegisterNumber start, end;
+   TR::DataType dt = node->getType();
+   if (dt.isIntegral() || dt.isAddress())
+      {
+      start = firstGPR;
+      end = lastGPR;
+      }
+   else if (dt.isFloatingPoint())
+      {
+      start = firstFPR;
+      end = lastFPR;
+      }
+   else if (dt.isVector())
+      {
+      start = firstVRF;
+      end = lastVRF;
+      }
+   else
+      TR_ASSERT_FATAL(false, "Unknown data type encountered when trying to pick a register for postGRA block splitting!");
```

Comment:
Let's add the datatype we encountered in the assert. I think there is a `toString` API you can use to extract the name.

Reply:
Fixed in https://github.com/eclipse/omr/commit/e75062fc5ab9ebb3996cc60947cb12fb2f027537